### PR TITLE
update boilerplate header and apply go fmt

### PR DESCRIPTION
**Pull request by cpanato**
- make all boilerplate header similar to each other
- update some go formating code
- add ci verify to validate the boilerplates
- add golangci-lint rules

sorry, big PR :(

more context slack 🧵 : https://sigstore.slack.com/archives/C01CQ3L3H8X/p1617957935102100

/assign @dlorenc @lukehinds

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -12,7 +27,7 @@ $(GENSRC): $(OPENAPIDEPS)
 
 # this exists to override pattern match rule above since this file is in the generated directory but should not be treated as generated code
 pkg/generated/restapi/configure_rekor_server.go: $(OPENAPIDEPS)
-	
+
```
Comment (bobcallaway) [inline Makefile]: There needs to be a tab here for the Makefile to be correct

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -1,49 +1,45 @@
...
-      # Lint it
-      - name: golangci-lint
-        uses: golangci/golangci-lint-action@v2.5.2
-        with:
-          version: v1.37.1
-      # Test It
       - name: Test
         run: go test -v ./...
-      # Gosec It
       - name: Run Gosec Security Scanner
```
Comment (bobcallaway) [inline .github/workflows/main.yml]: is this duplicative now that gosec is configured in the linting step?

**Review by bobcallaway — COMMENTED** [review]
generally looks good, just a couple comments

File: .github/workflows/main.yml (hunk 2 above)
Comment (cpanato) [reply .github/workflows/main.yml]: true, thanks for the catch

File: Makefile (hunk 1 above)
Comment (cpanato) [reply Makefile]: it works locally for me as is
```
$ make pkg/generated/restapi/configure_rekor_server.go
make: `pkg/generated/restapi/configure_rekor_server.go' is up to date.

```

Comment (cpanato) [reply Makefile]: but added back

Comment (cpanato) [reply Makefile]: thanks for your review @bobcallaway

**Review by bobcallaway — APPROVED** [review]
lgtm, thanks @cpanato

**Pull request merged**